packages/engine/src/renderer/components/AnimateScaleComponent.ts

```diff
@@ -63,18 +63,19 @@ export const AnimateScaleComponent = defineComponent({
       const transformComponent = getComponent(entity, TransformComponent)
       const originalScale = transformComponent.scale.clone()
 
-      animateScale(entity, originalScale.clone().multiplyScalar(sizeMultiplier.value), true)
+      animateScale(entity, originalScale.clone().multiplyScalar(sizeMultiplier.value))
 
       return () => {
-        animateScale(entity, originalScale, false)
+        if (!entityExists(entity)) return
+        animateScale(entity, originalScale)
       }
     }, [])
 
     return null
   }
 })
 
-const animateScale = (entity: Entity, newScale: Vector3, grow: boolean) => {
+const animateScale = (entity: Entity, newScale: Vector3) => {
```

Comment:
wonder when we lost `grow`. Haven't tested this is a minute. Hopefully it still shrinks.